Adafruit_MCP9808.cpp

```diff
@@ -46,9 +46,37 @@ Adafruit_MCP9808::Adafruit_MCP9808() {
     @brief  Setups the HW
 */
 /**************************************************************************/
-boolean Adafruit_MCP9808::begin(uint8_t addr) {
+
+bool Adafruit_MCP9808::begin(TwoWire *theWire)
+{
+  _wire = theWire;
+  _i2caddr = MCP9808_I2CADDR_DEFAULT;
+  return init();
+}
+
+bool Adafruit_MCP9808::begin(uint8_t addr)
+{
   _i2caddr = addr;
-  Wire.begin();
+  _wire = &Wire;
+  return init();
+}
+
+bool Adafruit_MCP9808::begin(uint8_t addr, TwoWire *theWire)
+{
+    _i2caddr = addr;
+  _wire = theWire;
+  return init();
+}
+
+bool Adafruit_MCP9808::begin(void)
+{
+    _i2caddr = MCP9808_I2CADDR_DEFAULT;
+  _wire = &Wire;
+  return init();
+}
+
+bool Adafruit_MCP9808::init(){
+  _wire -> begin();
```

Comment:
If you are passing in a specific wire instance, you may well have already called begin with required parameters. I don't think you should be calling it here again.

Reply:
but it works, i had a look into your [BME280 library](https://github.com/adafruit/Adafruit_BME280_Library/blob/master/Adafruit_BME280.cpp#L76) which i was inspired of and you doing exactly the same

Reply:
Looks good to me. 
If you use the default method and address it uses Wire and calls begin for you. 
If you want to set the clock rates, or use the same Wire for multiple sensors etc. you can pass in Wire or Wire1 etc. and call begin yourself.
Perhaps an example where you use 2 of the same sensor with different addresses would be useful to show this?